### ISSUE TITLE
Sonarqube 6.7.3 Issue - Build Breaker plugin

### ISSUE THREAD

**Issue by jbdundas** [adnovum/sonar-build-breaker#34]
Hi,

We are having an issue with the Build Breaker plugin 2.2.0 such that when an analysis, which is tied to a quality gate, is running, it breaks the build even though there are no issues related to the condition specified in the logs of it.

Description:-
---------------------
We have the following quality gate:-
![image](https://user-images.githubusercontent.com/4275521/39434597-053ca666-4c67-11e8-8407-ecaa7c6650b4.png)

As you can see, there are no code smells present as a condition here. Thus, there should be no check on this.

Next, we run the analysis for this project ( which is already tied to this project)
---------------------------------------------------------------------------------------
INFO: Analysis report uploaded in 161ms
INFO: ANALYSIS SUCCESSFUL, you can browse <SONAR_PROJECT_LINK>
INFO: Note that you will be able to access the updated dashboard once the server has processed the submitted analysis report
INFO: More about the report processing at http://SONAR_SERVER/api/ce/task?id=AWMH2d6biqBKdZ2nt0_e
INFO: Executing post-job org.sonar.plugins.buildbreaker.QualityGateBreaker
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Waiting for report processing to complete...
INFO: Quality gate status: ERROR
ERROR: Code Smells: 1 > 0
ERROR: [BUILD BREAKER] Project did not meet 1 conditions
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 34.833s
INFO: Final Memory: 85M/1821M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Project does not pass the quality gate.
	at org.sonar.plugins.buildbreaker.QualityGateBreaker.checkQualityGate(QualityGateBreaker.java:212)
	at org.sonar.plugins.buildbreaker.QualityGateBreaker.executeOn(QualityGateBreaker.java:117)
	at org.sonar.scanner.phases.PostJobsExecutor.execute(PostJobsExecutor.java:66)
	at org.sonar.scanner.phases.PostJobsExecutor.execute(PostJobsExecutor.java:55)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:94)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:180)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:288)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:283)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:261)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:110)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:74)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
ERROR: 
ERROR: Re-run SonarQube Scanner using the -X switch to enable full debug logging.
[Pipeline] }
----------------------------------------------------------------------

In short, even though the condition is not there, it is still catching the code smells and breaking the build.
INFO: Quality gate status: ERROR
ERROR: Code Smells: 1 > 0
ERROR: [BUILD BREAKER] Project did not meet 1 conditions
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE

Next, if we do onto the UI of the project dashboard, you can see that the Quality Gate is passed.
![image](https://user-images.githubusercontent.com/4275521/39434839-bb5979e2-4c67-11e8-856f-7a342655991f.png)

Environment Used:-
Sonarqube 6.7.3
Jenkins ver. 2.89.4
INFO: SonarQube Scanner 2.8
INFO: Java 1.8.0_77 Oracle Corporation (64-bit)
INFO: Linux 3.10.0-693.2.2.el7.x86_64 amd64

Can you please help us fix this issue?

Thanks,

**Comment by mibcadet** [comment]
Same here, if I'd know Java enough to make PR...

**Comment by sandro-h** [comment]
Hey @jbdundas, similar to the answer in #26: I'm not sure if this is still a problem for you in the meantime. SonarQube is at version 7.9 these days, and we've also just released a new version of the sonar-build-breaker which supports it.
I'll therefore close this issue, but feel free to comment if you still encounter such problems with newer versions.

**Comment by Sanjaykurra** [comment]
@jbdundas - Is this issue resolved as I see suggestion is to upgrade SonarQube. We are still on 6.7.3 and see this issue. Do we have a fix rather than upgrade?
